### PR TITLE
Generate doxygen documentation and publish on github pages + code coverage published on coveralls.io

### LINKED ISSUE #41: Api referece - documentation
Where can I find some api referece and some examples? for example: how to use with rsa public and private key

### PR DESCRIPTION
**Pull request by prince-chrismc**
Since this library is used as a part of API security I think its only fair to expect a high level of quality.

Having documentation more accessibly helps prevent implementation mistakes.
Code coverage will help highlight short comings where improvements can be made.

closes #41 

- [ ] add link to gp-pages in `README.md` or project
- [ ] setup coveralls.io for upstream

### CODE REVIEW

**Comment by prince-chrismc** [comment]
https://prince-chrismc.github.io/jwt-cpp/ for a preview

**Comment by prince-chrismc** [comment]
My intention was to separate these two items but github was not liking my attempt...

One potential issue is this uses GitHub action (instead of continuing with travis) the integration with from the new actions makes all of this a lot easier.

**Comment by Thalhammer** [comment]
I think its fine to have them as one PR.
I haven't worked with github actions a lot before (mainly because I also use Gitlab a lot).
Taking a look at the code coverage report (which is definitly a good thing to have it looks like most of the untested lines is either error handling or unused code (like std::string overloads for exceptions).

A doxygen documentation is also good, but I think we might need to start documenting stuff for it to be trully usefull.

Sidenote: I do intend to rewrite most of the internals to use std::error_code instead of exceptions, as well as providing non throwing overloads for all public functions.

File: README.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 [![Codacy Badge](https://api.codacy.com/project/badge/Grade/5f7055e294744901991fd0a1620b231d)](https://app.codacy.com/app/Thalhammer/jwt-cpp?utm_source=github.com&utm_medium=referral&utm_content=Thalhammer/jwt-cpp&utm_campaign=Badge_Grade_Settings)
 [![Build Status](https://travis-ci.com/Thalhammer/jwt-cpp.svg?branch=master)](https://travis-ci.com/Thalhammer/jwt-cpp)
+[![Coverage Status](https://coveralls.io/repos/github/prince-chrismc/jwt-cpp/badge.svg?branch=master)](https://coveralls.io/github/prince-chrismc/jwt-cpp?branch=master)
```
Comment (Thalhammer) [inline README.md:5]: This needs to be changed to point to the original repo onve merged (or right before).

**Comment by prince-chrismc** [comment]
I've only started using them recently, they have been a pleasure!

Error handling is certainly the big one, it might be worth looking into fuzzing to see how that goes.

I see your error_codes have been an idea for while, certainly useful :+1:

**Pull request merged**